### PR TITLE
Return complete Object List under Bucket when Delimiter is not specified

### PR DESCRIPTION
**Pull request by windkit**
### Description
When user does not specify `Delimiter`, complete object list under the bucket should be returned

### Related Issue
https://github.com/leo-project/leofs/issues/367

### CODE REVIEW

File: src/leo_gateway_s3_api.erl [hunk 1]
```diff
@@ -206,16 +206,26 @@ get_bucket(Req, Key, #req_params{access_key_id = AccessKeyId,
                                end
                        end,
     MaxKeys = case cowboy_req:qs_val(?HTTP_QS_BIN_MAXKEYS, Req) of
-                  {undefined, _} -> 1000;
+                  {undefined, _} -> ?HTTP_MAXKEYS_LIMIT;
                   {Val_2,     _} ->
                       try
-                          list_to_integer(binary_to_list(Val_2))
+                          case list_to_integer(binary_to_list(Val_2)) of
```
Comment (mocchira) [inline src/leo_gateway_s3_api.erl]: Using binary_to_integer is more efficient.

File: src/leo_gateway_s3_api.erl [hunk 2]
```diff
@@ -206,16 +206,26 @@ get_bucket(Req, Key, #req_params{access_key_id = AccessKeyId,
...
+                              MaxKeys1 when MaxKeys1 > ?HTTP_MAXKEYS_LIMIT ->
+                                  ?HTTP_MAXKEYS_LIMIT;
+                              MaxKeys1 ->
+                                  MaxKeys1
+                          end
```
Comment (mocchira) [inline src/leo_gateway_s3_api.erl]: Using `erlang:min `  less verbose.

**Comment by mocchira** [comment]
LGTM.
Please rebase to resolve conflict.

**Comment by windkit** [comment]
@mocchira Merged, please check again :)

**Comment by mocchira** [comment]
Thanks for fixing wired PR dependencies.
@yosukehara please merge if you're ok.

**Comment by yosukehara** [comment]
This request has been merged into the 1.4-branch. Thanks.

**Comment by yosukehara** [comment]
A related issue is leo-project/leofs/issues/367

**Pull request merged**